pentasharp/Controllers/BusCompanyController.cs

```diff
@@ -2,147 +2,94 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
 using pentasharp.Data;
+using pentasharp.Interfaces;
 using pentasharp.Models.Entities;
+using pentasharp.Models.Enums;
+using pentasharp.Models.Utilities;
 using pentasharp.ViewModel.Bus;
 using WebApplication1.Filters;
 
 namespace WebApplication1.Controllers
 {
-    [Route("api/BusCompany")]
+    [Route("Admin/BusCompany")]
     [ServiceFilter(typeof(AdminOnlyFilter))]
     public class BusCompanyController : Controller
     {
+        private readonly IBusCompanyService _companyService;
         private readonly AppDbContext _context;
+        private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly IMapper _mapper;
 
-        public BusCompanyController(AppDbContext context, IMapper mapper)
+        public BusCompanyController(IBusCompanyService companyService, IHttpContextAccessor httpContextAccessor, AppDbContext context, IMapper mapper)
         {
+            _companyService = companyService;
+            _httpContextAccessor = httpContextAccessor;
             _context = context;
             _mapper = mapper;
         }
 
-        [HttpGet("Bus")]
-        public IActionResult Add()
+        [HttpGet("GetBusCompanyUsers")]
+        public async Task<IActionResult> GetBusCompanyUsersAsync()
         {
-            var companies = _context.BusCompanies.ToList();
+            var users = await _companyService.GetBusCompanyUsersAsync();
```

Comment:
if users are null should be checked in service, and you should check for when users are null to return or log an error/information. Move this logic and all other logic to its service

Reply:
why was this comment resolved but changes are not done?

Reply:
I don't know why I thought that I improved it